runners/java-fn-execution/src/main/java/org/apache/beam/runners/fnexecution/environment/DockerEnvironmentFactory.java

```diff
@@ -225,6 +235,11 @@ private static ServerFactory getServerFactory() {
 
   /** Provider for DockerEnvironmentFactory. */
   public static class Provider implements EnvironmentFactory.Provider {
+    private final boolean retain;
```

Comment:
Would rename this to `retainDockerContainer`.

Reply:
Sure!